assets/js/blocks/featured-product/block.js

```diff
@@ -103,17 +106,34 @@ class FeaturedProduct extends Component {
 		const { productId } = this.props.attributes;
 		if ( ! productId ) {
 			// We've removed the selected product, or no product is selected yet.
-			this.setState( { product: false, loaded: true } );
+			this.setState( { product: false, loaded: true, error: false } );
 			return;
 		}
 		apiFetch( {
 			path: `/wc/blocks/products/${ productId }`,
 		} )
 			.then( ( product ) => {
-				this.setState( { product, loaded: true } );
+				this.setState( { product, loaded: true, error: false } );
 			} )
-			.catch( () => {
-				this.setState( { product: false, loaded: true } );
+			.catch( ( apiError ) => {
+				const error = {
+					retry: this.debouncedGetProduct,
+				};
+
+				if ( isObject( apiError ) ) {
+					error.message = (
+						<span dangerouslySetInnerHTML={
+							{
+								__html: escapeHTML( __( 'The following error was returned from the API', 'woo-gutenberg-products-block' ) ) + ': <br/><code>' + escapeHTML( apiError.message ) + '</code>',
+							}
+						} />
+					);
+				} else {
+					error.message = escapeHTML( __( 'An unknown error occured which prevented the block from being updated.', 'woo-gutenberg-products-block' ) );
```

Comment:
`escapeHTML` shouldn't be needed here if we trust translations. Oh, and there was a typo in _occur**r**ed_. :slightly_smiling_face: 
```suggestion
					error.message = __( 'An unknown error occurred which prevented the block from being updated.', 'woo-gutenberg-products-block' );
```

Reply:
@Aljullu We shouldn't really trust them - in core you have esc_html__ etc but these are not available in Gutenberg.

Reply:
Ok, makes sense, thank you. But in any case, we are no longer rendering it inside `dangerouslySetInnerHTML`, so it should be safe to remove `escapeHTML`, right?

Reply:
yup when thats changed